File: DAY 9/solutions.java/Maximum Sum of Distinct Subarrays With Length K.java
import java.util.*;

class Solution {
    public long maximumSubarraySum(int[] nums, int k) {

        int left = 0;
        long sum = 0;
        long answer = 0;

        Map<Integer, Integer> seen = new HashMap<>();

        for (int right = 0; right < nums.length; right++) {

           
            sum += nums[right];

            seen.put(nums[right],
                    seen.getOrDefault(nums[right], 0) + 1);

            
            if (right - left + 1 == k) {

               
                if (seen.size() == k) {
                    answer = Math.max(answer, sum);
                }

              
                seen.put(nums[left],
                        seen.get(nums[left]) - 1);

                if (seen.get(nums[left]) == 0) {
                    seen.remove(nums[left]);
                }

                sum -= nums[left];

                
                left++;
            }
        }

        return answer;
    }
}

File: DAY 9/solutions.py/Maximum Sum of Distinct Subarrays With Length K.py
class Solution:
    def maximumSubarraySum(self, nums: List[int], k: int) -> int:

        left = 0
        sum = 0
        answer = 0
        seen = {}

        for right in range(len(nums)):

            sum += nums[right]

            if nums[right] in seen:
                seen[nums[right]] += 1
            else:
                seen[nums[right]] = 1

            if right - left + 1 == k:

                if len(seen) == k:
                    answer = max(answer, sum)

                seen[nums[left]] -= 1
                sum -= nums[left]

                if seen[nums[left]] == 0:
                    del seen[nums[left]]

                left += 1

        return answer